packages/errors/src/error-codes.ts

```diff
@@ -21,6 +21,10 @@ export enum ErrorCode {
   // transaction
   TRANSACTION_FAILED = 'transaction-failed',
 
+  // contract
+  ONLY_ONE_HEAP_TYPE_CALL_ALLOWED = 'only-one-heap-type-call-allowed',
```

Comment:
I think that both of these error codes can be put into an `INVALID_MULTICALL` error code, and the message of the error would give details on what's the cause of the error:
```ts
export enum ErrorCode {
...

/**
* Gets thrown on the client side when a multicall wasn't properly configured.
*/
INVALID_MULTICALL = 'invalid-multicall',

...
}
```
Note for all:
We should give special attention to these error codes and try to encapsulate multiple failure modes into one error code, where it makes sense. Then the `message` would be used to explain the reason in detail. Otherwise, we might find ourselves in a situation where we've got an error code per failure mode where the `message`s don't add anything substantial of description to the error code, in which case one could rightfully ask, "Why do we need  the `message` at all?"